infra/build_deploy_config.py

```diff
@@ -0,0 +1,70 @@
+#!/usr/bin/env python
+
+import os
+import boto3
+import argparse
+
+
+infra_root = os.path.abspath(os.path.dirname(__file__))
+
+
+parser = argparse.ArgumentParser(description=__doc__)
+parser.add_argument("component")
+args = parser.parse_args()
+
+terraform_variable_template = """
+variable "{name}" {{
+  default = "{val}"
+}}
+"""
+
+terraform_backend_template = """# Auto-generated during infra build process.
+# Please edit infra/build_deploy_config.py directly.
+terraform {{
+  backend "s3" {{
+    bucket = "{bucket}"
+    key = "{comp}-{stage}.tfstate"
+    region = "{region}"
+    {profile_setting}
+  }}
+}}
+"""
+
+terraform_providers_template = """# Auto-generated during infra build process.
+# Please edit infra/build_deploy_config.py directly.
+provider aws {{
+  region = "{aws_region}"
+}}
+"""
+
+env_vars_to_infra = [
+    "AWS_DEFAULT_OUTPUT",
+    "AWS_DEFAULT_REGION",
+    "DCP_DEPLOYMENT_STAGE"
+]
+
+with open(os.path.join(infra_root, args.component, "backend.tf"), "w") as fp:
+    caller_info = boto3.client("sts").get_caller_identity()
+    if os.environ.get('AWS_PROFILE'):
+        profile = os.environ['AWS_PROFILE']
+        profile_setting = f'profile = "{profile}"'
+    else:
+        profile_setting = ''
+    fp.write(terraform_backend_template.format(
+        bucket=os.environ['DCP_TERRAFORM_BACKEND_BUCKET_TEMPLATE'].format(account_id=caller_info['Account']),
+        comp=args.component,
+        stage=os.environ['DCP_DEPLOYMENT_STAGE'],
+        region=os.environ['AWS_DEFAULT_REGION'],
+        profile_setting=profile_setting,
+    ))
+
+with open(os.path.join(infra_root, args.component, "variables.tf"), "w") as fp:
```

Comment:
Please try to use descriptive names instead of a few letters representing the object.
`fp` -> `terraform_file_pointer` for example.

Reply:
I see no problem with this, name since the name is not used outside of this context and the function is short. `fp` is also commonly used in many languages to mean file pointer.